rdrf/explorer/views.py

```diff
@@ -1,38 +1,38 @@
-from datetime import datetime
-from itertools import product
 import logging
 import re
+from datetime import datetime
+from itertools import product
 from tempfile import NamedTemporaryFile
 
 from csp.decorators import csp_update
 from django.conf import settings
-
 from django.core.exceptions import PermissionDenied
-from django.urls import reverse
 from django.http import HttpResponse, FileResponse, HttpResponseRedirect, JsonResponse, StreamingHttpResponse
 from django.shortcuts import get_object_or_404, render, redirect
+from django.urls import reverse
 from django.views.generic.base import View
 
 from explorer import __version__
-from .forms import QueryForm
-from .models import Query
-from .utils import DatabaseUtils
+from rdrf.helpers.utils import models_from_mongo_key, is_delimited_key, BadKeyError, cached
+from rdrf.helpers.utils import mongo_key_from_models, check_suspicious_sql
 from rdrf.models.definition.models import Registry
 from rdrf.models.definition.models import RegistryForm
 from rdrf.models.definition.models import Section
-from registry.groups.models import WorkingGroup
 from rdrf.security.mixins import SuperuserRequiredMixin
-from rdrf.services.io.reporting.spreadsheet_report import SpreadSheetReport
 from rdrf.services.io.reporting.reporting_table import ReportingTableGenerator
-
-from rdrf.helpers.utils import models_from_mongo_key, is_delimited_key, BadKeyError, cached
-from rdrf.helpers.utils import mongo_key_from_models, check_suspicious_sql
+from rdrf.services.io.reporting.spreadsheet_report import SpreadSheetReport
+from rdrf.views.decorators.report_decorators import is_legacy_reports_enabled
+from registry.groups.models import WorkingGroup
+from .forms import QueryForm
+from .models import Query
+from .utils import DatabaseUtils
 
 logger = logging.getLogger(__name__)
 
 
 class MainView(View):
 
+    @is_legacy_reports_enabled
```

Comment:
`rpc_reporting_command` in `rdrf/rdrf/services/rpc/rpc_commands.py` should also be restricted unless legacy reports are enabled

Reply:
Still to do.

Reply:
Done.